### PR TITLE
Inline icons

### LINKED ISSUE #79: Add Icon via Paragraph
<a href="https://github.com/jauyong"></img></a> **Issue by [jauyong](https://github.com/jauyong)**
_Wednesday Feb 17, 2021 at 18:08 GMT_
_Originally opened as https://github.com/xwp/material-design-wp-plugin/issues/540_

----

## Feature description

As a site editor, I'd like to be able to add a Materical Icon while not using an `icon button`

See original slack conversation with Yasmine [here](https://xwp.slack.com/archives/CSS5LFZJB/p1613521802114500). 

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

1. I am able to add an icon within a paragraph block without using an 'icon button' 
2. It should not have the tooltip
3. It should not have the ripple effect

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

### PR DESCRIPTION
**Pull request by emeaguiar**
## Summary

Fixes  #79 

## Checklist

- [ ] My pull request is addressing an [open issue](https://github.com/material-components/material-design-for-wordpress/issues) (please create one otherwise).
- [ ] My code is tested and passes existing [tests](https://github.com/material-components/material-design-for-wordpress/blob/main/contributing/engineering.md#tests).
- [ ] My code follows the [Contributing Guidelines](https://github.com/material-components/material-design-for-wordpress/blob/main/contributing.md) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -73,7 +73,7 @@
     "@wordpress/jest-puppeteer-axe": "1.6.0",
     "@wordpress/plugins": "^3.1.0",
     "@wordpress/postcss-themes": "2.3.0",
-    "@wordpress/rich-text": "3.24.1",
+    "@wordpress/rich-text": "^4.1.2",
```
Comment (PatelUtkarsh) [inline package.json:76]: Not related to this line: but should we keep version 2 in `package-lock.json` file?

File: plugin/assets/src/block-editor/formats/icon/anchor-ref-popover.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/**
```
Comment (PatelUtkarsh) [inline plugin/assets/src/block-editor/formats/icon/anchor-ref-popover.js:1]: Missing copyright.

**Review by PatelUtkarsh — COMMENTED** [review]
Nice work @emeaguiar! Left some minor feedback.

E2E with 5.2 WP seems to be breaking, Likely because of rich-text dependency missing? Do we need to polyfill or should we upgrade the minimum WordPress version?

**Comment by PatelUtkarsh** [comment]
Few things I noticed on quick testing: 

1. When I copy past text with icon block it removes span and shows the icon text. 

image

2. When making text large icon stays small on the frontend and backend.

image

3. On delete of icon it creates extra spaces.

https://user-images.githubusercontent.com/5015489/126347527-94a85aa7-3d0a-4ddc-9676-1ec8bbdab6f4.mp4

File: theme/inc/widgets/class-wp-widget-pages.php [hunk 3]
```diff
@@ -79,7 +78,6 @@ public function widget( $args, $instance ) {
 					'title_li'    => '',
 					'echo'        => 0,
 					'sort_column' => $sortby,
-					'exclude'     => $exclude,
```
Comment (PatelUtkarsh) [inline theme/inc/widgets/class-wp-widget-pages.php:82]: Is this error related to the latest WordPress or something else?

**Comment by emeaguiar** [comment]
I'm not really seeing issue 1, issue 2 is fixed, it nows inherits the properties of the paragraph. Working on 3


https://user-images.githubusercontent.com/1379732/126812656-bd72dc14-847f-444d-9397-ddd91fb3b231.mp4

**Comment by PatelUtkarsh** [comment]
@googlebot I consent.

**Review by PatelUtkarsh — DISMISSED** [review]
PR looks good, Nice work! ✅ 

On removing icon with backspace adds some whitespace - selection of icon and delete works fine.

**Comment by PatelUtkarsh** [comment]
@googlebot I consent.

**Review by PatelUtkarsh — APPROVED** [review]

**Comment by emeaguiar** [comment]
@googlebot I consent.

**Comment by pbearne** [comment]
@googlebot I consent

**Pull request merged**